python/pyspark/mllib/classification.py

```diff
@@ -207,7 +207,7 @@ def train(cls, data, iterations=100, initialWeights=None, regParam=0.01, regType
         """
         def train(rdd, i):
             return callMLlibFunc("trainLogisticRegressionModelWithLBFGS", rdd, int(iterations), i,
-                                 float(regParam), str(regType), bool(intercept), int(corrections),
+                                 float(regParam), regType, bool(intercept), int(corrections),
```

Comment:
The difference is the error message. If we use `str(...)`, the Java function call would be successful and inside the Java function, we say "type" is not recognized.

If we don't use `str(..)`, if users input something like `1`, `2`. Py4j will throw an error saying there is no Java function matching the input signature, which usually confuses users.

Anyway, this is a minor issue for `regType`.